### ISSUE TITLE
regression: ICE no resolution found for import

### ISSUE THREAD

**Issue by BoxyUwU** [rust-lang/rust#147966]
- https://crater-reports.s3.amazonaws.com/beta-1.91-2/beta-2025-09-21/gh/gruberb.ohttp-gateway/log.txt
- https://crater-reports.s3.amazonaws.com/beta-1.91-3/beta-2025-09-21/reg/jitash-bdk-0.28.1/log.txt

```
[INFO] [stdout] error: internal compiler error: no resolution for an import
[INFO] [stdout]  --> tests/config_handler_tests.rs:1:5
[INFO] [stdout]   |
[INFO] [stdout] 1 | use hyper::StatusCode;
[INFO] [stdout]   |     ^^^^^^^^^^^^^^^^^
[INFO] [stdout]   |
[INFO] [stdout] note: delayed at compiler/rustc_ast_lowering/src/lib.rs:760:24
```

**Comment by frogtd** [comment]
This does compile on stable on Linux, and ICEs on beta, but if you remove everything but the two conflicting deps, it does give the error about duplicated rlib's, both on stable and beta:
```
error[E0464]: multiple candidates for `rlib` dependency `hyper` found
 --> tests/config_handler_tests.rs:1:5
  |
1 | use hyper::StatusCode;
  |     ^^^^^
  |
  = note: candidate #1: /home/ash/Downloads/ohttp-gateway/target/debug/deps/libhyper-08a7d65fae167315.rlib
  = note: candidate #2: /home/ash/Downloads/ohttp-gateway/target/debug/deps/libhyper-a6c863e938beb4b9.rlib
```

**Comment by theemathas** [comment]
My guess is: There's a preexisting bug in stable that fails to detect duplicate rlibs, but only in some conditions, and not on windows. This bug was previously hidden, and manifested as an ICE on beta due to some additional check performed.

**Comment by petrochenkov** [comment]
> My guess is: There's a preexisting bug in stable that fails to detect duplicate rlibs, but only in some conditions, and not on windows. This bug was previously hidden, and manifested as an ICE on beta due to some additional check performed.

This is true, the root issue is in cstore, `(maybe_)process_path_extern` returns different results when called twice for the name `hyper`.
I continue investigating.

**Comment by petrochenkov** [comment]
> [@petrochenkov](https://github.com/petrochenkov) On stable, it errors on windows, but compiles on linux. So it's a "compiles -> ICE" situation.

In any case the situation when this ICEs is better than when it compiles (and ideally it should error, in a target independent way).
